.github/workflows/xp.yml

```diff
@@ -47,6 +48,7 @@ jobs:
         run: make lint-python
 
   lint-go:
+    if: github.head_ref != 'refs/heads/main'
```

Comment:
Doesn't this mean we will run this step for tags too? Should we split the commit to non-main branch from the release steps (release from `main` or from `tag`) into 2 files?

Reply:
Are you proposing for `release-rules` and `release` steps to be in a separate file? And that all `lint` + `test` steps to be only for PRs?

Reply:
Not exactly.. Something like:
1. test.yaml - this one has all the lint / test steps for the "commit to non-main branch" event (cmiiw but with the current rule `if: github.head_ref != 'refs/heads/main'`, the lint / test will also be run for tag creation, which we want to avoid).
2. release.yaml - this one has the release steps (build & publish) for the main branch run / tag create / manual workflow dispatch events.

If we split it this way, we can have 2 separate files with the condition for triggering the workflow with the `on` tag rather than the `if` spec per task.

Reply:
>the lint / test will also be run for tag creation, which we want to avoid

Oh then I guess I misunderstood, so CMIIW, but we want lint/test to run in non-main branch, so that we can test the changes in the event we cherry-pick commits prior to cutting a new tag right?

> release.yaml - this one has the release steps (build & publish) for the main branch run / tag create / manual workflow dispatch events.

So are you suggesting to do away with the current `workflow_call` trigger?